packages/front-end/components/Settings/Team/PendingMemberList.tsx

```diff
@@ -0,0 +1,157 @@
+            <th>Email</th>
+            <th>Date Joined</th>
+            <th>{project ? "Project Role" : "Global Role"}</th>
+            {!project && <th>Project Roles</th>}
+            {environments.map((env) => (
+              <th key={env.id}>{env.id}</th>
+            ))}
+            <th />
+            <th style={{ width: 50 }} />
+          </tr>
+        </thead>
+        <tbody>
+          {pendingMembers.map((member) => {
+            const roleInfo =
+              (project &&
+                member.projectRoles?.find((r) => r.project === project)) ||
+              member;
+            return (
+              <tr key={member.id}>
+                <td>{member.name}</td>
+                <td>{member.email}</td>
+                <td>{member.dateCreated && datetime(member.dateCreated)}</td>
+                <td>{roleInfo.role}</td>
+                {!project && (
+                  <td className="col-3">
+                    {member.projectRoles.map((pr) => {
+                      const p = projects.find((p) => p.id === pr.project);
+                      if (p?.name) {
+                        return (
+                          <div key={`project-tags-${p.id}`}>
+                            <ProjectBadges
+                              projectIds={[p.id]}
+                              className="badge-ellipsis align-middle font-weight-normal"
+                            />
+                            — {pr.role}
+                          </div>
+                        );
+                      }
+                      return null;
+                    })}
+                  </td>
+                )}
+                {environments.map((env) => {
+                  const access = roleHasAccessToEnv(roleInfo, env.id);
+                  return (
+                    <td key={env.id}>
+                      {access === "N/A" ? (
+                        <span className="text-muted">N/A</span>
+                      ) : access === "yes" ? (
+                        <FaCheck className="text-success" />
+                      ) : (
+                        <FaTimes className="text-danger" />
+                      )}
+                    </td>
+                  );
+                })}
+                <td>
+                  <button
+                    className="btn btn-outline-success px-2"
+                    onClick={async () => {
+                      await apiCall(`/member/${member.id}/approve`, {
+                        method: "POST",
+                      });
+                      mutate();
+                    }}
+                  >
+                    <FaUserCheck /> Approve
+                  </button>
+                </td>
+                <td>
+                  <MoreMenu>
+                    <button
+                      className="dropdown-item"
+                      onClick={(e) => {
+                        e.preventDefault();
+                        setRoleModal(member.id);
```

Comment:
And here, we would just need to pass `member` instead

Reply:
This is a good insight! Added your suggestion, thanks!